### PR TITLE
Maxlang/ui number viz

### PR DESCRIPTION
**Pull request by maxlang**
Move stats into  "visualizations"

screen shot 2015-10-13 at 1 58 32 pm

[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/2829)

### CODE REVIEW

**Comment by maxlang** [comment]
@mrtracy I took out the width/height logic for now

**Comment by maxlang** [comment]
I'll fix up the commit history if you like these changes.

**Comment by mrtracy** [comment]
Review status: 0 of 17 files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---

<sup>**[ui/ts/components/visualizations/bar.ts, line 9 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/2829#-K0c0i68W5W5txcw65OT)** ([raw file](https://github.com/cockroachdb/cockroach/blob/fd444a0157ca2106818ba30b5c2950e304b02a57/ui/ts/components/visualizations/bar.ts#L9)):</sup>
Can we at least put a todo comment on these skeletons?  I don't really like checking in unused code, this could easily be recreated when needed.


---

<sup>**[ui/ts/components/visualizations/number.ts, line 35 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/2829#-K0c0vWF_zmJ7-lNYVF6)** ([raw file](https://github.com/cockroachdb/cockroach/blob/fd444a0157ca2106818ba30b5c2950e304b02a57/ui/ts/components/visualizations/number.ts#L35)):</sup>
If there is only one contained child element in an m(), you do not need to pass it as a list.


---

<sup>**[ui/ts/components/visualizations/visualizations.ts, line 23 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/2829#-K0c1B-8XMqPwsEGmyPo)** ([raw file](https://github.com/cockroachdb/cockroach/blob/fd444a0157ca2106818ba30b5c2950e304b02a57/ui/ts/components/visualizations/visualizations.ts#L23)):</sup>
I still do not like this pattern, but it's been a day and I can't think of anything obviously better.  We have a partial application problem; I would like to pass down some sort of complete component into the VisualizationWrapper, and have the VisualizationWrapper (which is like a layout component) adjust properties on the component before rendering it.  

Unforunately, there's no clean way that I can think of to do this in mithril, without doing what you originally did; passing in both the class and the incomplete parameters, and calling m.component() in the visualization wrapper.  This model does not allow you to compose components well; however, it's only one level for now, so it's fine.  I don't like it, we'll fix it later.


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/2829)

**Comment by maxlang** [comment]
Review status: 0 of 17 files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---

<sup>**[ui/ts/components/visualizations/bar.ts, line 9 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/2829#-K0c0i68W5W5txcw65OT)** ([raw file](https://github.com/cockroachdb/cockroach/blob/fd444a0157ca2106818ba30b5c2950e304b02a57/ui/ts/components/visualizations/bar.ts#L9)):</sup>
I'll add TODOs, since at this point it doesn't hurt to have them in (doesn't affect the generated code)


---

<sup>**[ui/ts/components/visualizations/number.ts, line 35 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/2829#-K0c0vWF_zmJ7-lNYVF6)** ([raw file](https://github.com/cockroachdb/cockroach/blob/fd444a0157ca2106818ba30b5c2950e304b02a57/ui/ts/components/visualizations/number.ts#L35)):</sup>
Good point


---

<sup>**[ui/ts/components/visualizations/visualizations.ts, line 23 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/2829#-K0c1B-8XMqPwsEGmyPo)** ([raw file](https://github.com/cockroachdb/cockroach/blob/fd444a0157ca2106818ba30b5c2950e304b02a57/ui/ts/components/visualizations/visualizations.ts#L23)):</sup>
Sounds good. If you think of a better pattern we can always adjust.


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/2829)

**Comment by mrtracy** [comment]
LGTM

File: ui/ts/components/visualizations/number.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+  export module NumberVisualization {
+    export function controller(): any {
+    }
+
+    export function view(ctrl: any, info: NumberVisualizationConfig): MithrilVirtualElement {
+
+      formatFn = info.formatFn || d3.format(info.format || ".3s");
+
+      let data: VisualizationData = _.isArray(info.data) ? info.data[0] : info.data;
```
Comment (tamird) [inline ui/ts/components/visualizations/number.ts]: why would info.data ever be an array? isn't it strictly a `NumberVisualizationData`?

File: ui/ts/components/visualizations/visualizations.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+/// <reference path="./pie.ts" />
+
+// Author: Max Lang (max@cockroachlabs.com)
+
+module Visualizations {
+  "use strict";
+  import MithrilVirtualElement = _mithril.MithrilVirtualElement;
+
+  export interface VisualizationData {
+    key?: string;
```
Comment (tamird) [inline ui/ts/components/visualizations/visualizations.ts]: this field appears to be unused

File: ui/ts/components/visualizations/visualizations.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    value: number;
+  }
+
+  export interface VisualizationWrapperInfo {
+    title?: string;
```
Comment (tamird) [inline ui/ts/components/visualizations/visualizations.ts]: this field doesn't need to be optional

**Pull request merged**